server/CGameHandler.cpp

```diff
@@ -4338,24 +4338,32 @@ void CGameHandler::useChargedArtifactUsed(const ObjectInstanceID & heroObjectID,
 	if(vstd::contains(hero->getSpellsInSpellbook(), spellID))
 		return;
```

Comment:
Found at least one case where this check is insufficient - use `vcmiistari` to get all spells and cast something.